### DISCUSSION TITLE
Ghostty build.zig bzip2 not found on Ubuntu 24.04

### DISCUSSION THREAD

**Discussion by anjandev** [ghostty-org/ghostty#4143] [Q&A]
I tried building ghostty v1.0.0 on Ubuntu 24.04 and zig could not find the bzip2 library unless I made the following changed this line: https://github.com/ghostty-org/ghostty/blob/v1.0.0/build.zig#L1067

change from

 `            step.linkSystemLibrary2("bzip2", dynamic_link_opts); `

change to

 `            step.linkSystemLibrary2("bz2", dynamic_link_opts); `

Is this a known issue? Reporting here just in case someone else runs into this problem.

**Comment by ghost** [comment]
I had to run `sudo ln -s /usr/lib/x86_64-linux-gnu/libbz2.so /usr/lib/x86_64-linux-gnu/libbzip2.so` to fix this up outside the ghostty code.

I have `main` checked out. When I try to run the compile for packages, I get an error
```
error: unable to find dynamic system library 'bzip2' using strategy 'mode_first'. searched paths:
  /usr/lib/x86_64-linux-gnu/libbzip2.so
  ...
```
Which led me to trying that `ln` command.

"bz2" seems to be correct, but yet things build correctly with "bzip2" when you run zig build, but when you use the `compile for packages` example, the missing library error pops up. 

Seems to be related to `--system` and just removing that line `step.linkSystemLibrary2("bzip2", dynamic_link_opts);` seems to be the correct fix. It will build both with and without --system after you remove that line.

It works with `bz2` because I think that is the correct string, but when you run zig build without the --system option, I think it branches into the `else` statement, and the erroneous bzip2 is never even used.

```
        if (b.systemIntegrationOption("freetype", .{})) {
            //i think remove below line or at least change it to "bz2" if it really is needed in some cases
            //step.linkSystemLibrary2("bzip2", dynamic_link_opts);
            step.linkSystemLibrary2("freetype2", dynamic_link_opts);
        } else {
            step.linkLibrary(freetype_dep.artifact("freetype"));
            try static_libs.append(freetype_dep.artifact("freetype").getEmittedBin());
        }
```
